apm-agent-plugins/apm-opentelemetry/apm-opentelemetry-plugin/src/main/java/co/elastic/apm/agent/opentelemetry/sdk/OTelSpanBuilder.java

```diff
@@ -86,13 +91,14 @@ public SpanBuilder setNoParent() {
 
     @Override
     public SpanBuilder addLink(SpanContext spanContext) {
-        addLinkLogger.warn("The addLink API is not supported at the moment");
+        links.add(((OTelSpanContext) spanContext).getElasticTraceContext());
         return this;
     }
 
     @Override
-    public SpanBuilder addLink(SpanContext spanContext, Attributes attributes) {
-        addLinkLogger.warn("The addLink API is not supported at the moment");
+    public SpanBuilder addLink(SpanContext spanContext, Attributes attributes1) {
+        addLink(spanContext);
+        attributes1.forEach((AttributeKey<?> k, Object v) -> attributes.put((AttributeKey<? super Object>) k, (Object) v));
```

Comment:
Is that the intention? In the OTel javadoc it only says: "the attributes of the Link". I don't know what this means - are those intended to be added to the built span attributes?

Reply:
These shouldn't be added to the span attributes as they're scoped to a particular span link. However, we don't have support for span link attributes (an no OTel auto instrumentation seems to use span link attributes yet). So we can just ignore the span link attributes for now.

Reply:
In this case I think this method should revert to the API not supported rather than add the link and ignore the attributes. That way it's clear to the user they shouldn't use it

Reply:
The only down side to that is that users the use it currently with `null` or empty attributes won't be able to migrate to using us.
What about logging a warning once, saying these attributes will be ignored?

Reply:
Yes, agree. I went with logging the warning every time, it's what we did previously and better to make that clear alongside the call when investigating